extensions/assistant/deployment-spi/src/main/java/io/quarkus/assistant/deployment/spi/AssistantBuildItem.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.assistant.deployment.spi;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * If available, a handle on the Assistant
+ *
+ * This is intended for use in dev mode to enable AI-enhanced development
+ */
+public final class AssistantBuildItem extends SimpleBuildItem {
```

Comment:
Can you add into the javadoc that the user is not intended to produce this BI and also, the consumption must be using `Optional` (meaning the the build step won't be called if not present)